lib/reader/src/parser.rs

```diff
@@ -903,6 +931,68 @@ impl<'a> Parser<'a> {
         // Collect comments for the next instruction to be allocated.
         self.gather_comments(ctx.function.dfg.next_inst());
 
+        let mut encoding: Option<Encoding> = None;
+        let mut result_registers: Option<Vec<&'a str>> = None;
+
+        // encoding ::= "[" (encoding_literal | "-") , (result_registers | "-") "]"
+        if self.optional(Token::LBracket) {
+            let mut encoding_literal: Option<(&'a str, u16)> = None;
+
+            // encoding_literal ::= Identifier HexSequence
+            if let Some(Token::Identifier(recipe)) = self.token() {
+                self.consume();
+
+                if let Some(Token::HexSequence(bits_str)) = self.token() {
+                    self.consume();
+
+                    // The only error we anticipate from this parse is overflow, the lexer should
+                    // already have ensured that the string doesn't contain invalid characters, and
+                    // isn't empty or negative.
+                    if let Ok(bits) = u16::from_str_radix(bits_str, 16) {
+                        encoding_literal = Some((recipe, bits));
+                    } else {
+                        return err!(self.loc, "the hex sequence given overflows the u16 type");
+                    }
```

Comment:
Maybe even a `match_hex16() -> Result<u16>` method? See the existing `match_uimm8()`.